examples/platforms/nrf52840/radio.c

```diff
@@ -679,6 +699,12 @@ void nrf_802154_received_raw(uint8_t *p_data, int8_t power, uint8_t lqi)
     receivedFrame->mInfo.mRxInfo.mMsec = timestamp / US_PER_MS;
     receivedFrame->mInfo.mRxInfo.mUsec = timestamp - receivedFrame->mMsec * US_PER_MS;
 #endif
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+    // Get the timestamp when the SFD was received.
+    uint32_t offset =
+        (int32_t)otPlatAlarmMicroGetNow() - (int32_t)nrf_802154_first_symbol_timestamp_get(time, p_data[0]);
+    receivedFrame->mIeInfo->mTimestamp = otPlatTimeGet() - offset;
```

Comment:
@chshu Isn't otPlatTimeGet() equal to otPlatAlarmMicroGetNow() ? It seems this effectively does:

```
receivedFrame->mIeInfo->mTimestamp = (int32_t)nrf_802154_first_symbol_timestamp_get(time, p_data[0]);
```

Reply:
```otPlatTimeGet()``` returns 64bits time, while ```otPlatAlarmMicroGetNow()``` returns 32bits time (may wrap):
* `nrf_802154_first_symbol_timestamp_get()` returns the 32bits timestamp when the first symbol was received;
* `offset` is the time elapsed since first symbol was received till now;
* mTimestamp is the 64bits timestamp when the first symbol was received.